src/ffi.h

```diff
@@ -0,0 +1,15 @@
+#include <algorithm>
+#include <memory>
+
+#include "cxx.h"
+#if INCLUDE_RUST_HEADERS
+// For some unknown reason, the definition of rust::Box is in this particular header:
+#include "parse_constants.rs.h"
+#endif
+
+template <typename T>
+std::shared_ptr<T> box_to_shared_ptr(rust::Box<T> &&value) {
+    T *ptr = value.into_raw();
```

Comment:
can  we consolidate this with `src/rustffi.h`?

Reply:
There is no `rustffi.h` :)

Reply:
I might add it, and there is `rustffi.cpp` which has definitions for  UniquePtr<CxxWString>. Question is whether these two things go into the same bucket. I think yes.